src/guides/metamask-api/index.md

```diff
@@ -0,0 +1,1094 @@
+              Install Metamask
+            </a>
+          </Link>
+        )}
+
+        {isConnected && (
+          <div className="flex  w-full justify-center space-x-2">
+            <button
+              onClick={handleAddUsdc}
+              className="mt-8 inline-flex w-full items-center justify-center rounded-md border border-transparent bg-ganache text-white px-5 py-3 text-base font-medium  sm:w-auto"
+            >
+              {status === "loading" ? <Loading /> : "Add Token"}
+            </button>
+            <button
+              onClick={handleDisconnect}
+              className="mt-8 inline-flex w-full items-center justify-center rounded-md border border-transparent bg-ganache text-white px-5 py-3 text-base font-medium  sm:w-auto"
+            >
+              Disconnect
+            </button>
+          </div>
+        )}
+
+      </div>
+    </div>
+  );
+}
+```
+
+Finally, we will update our `pages/index.tsx` file with a `useEffect` to wrap all of these final changes up.
+
+Update `pages/index/tsx`
+
+```typescript
+import type { NextPage } from "next";
+import { useEffect } from "react";
+import Wallet from "../components/Wallet";
+import { useListen } from "../hooks/useListen";
+import { useMetamask } from "../hooks/useMetamask";
+
+const Home: NextPage = () => {
+  const { dispatch } = useMetamask();
+  const listen = useListen();
+
+  useEffect(() => {
+    if (typeof window !== undefined) {
+      // start by checking if window.ethereum is present, indicating a wallet extension
+      const ethereumProviderInjected = typeof window.ethereum !== "undefined";
+      // this could be other wallets so we can verify if we are dealing with metamask
+      // using the boolean constructor to be explecit and not let this be used as a falsy value (optional)
+      const isMetamaskInstalled =
+        ethereumProviderInjected && Boolean(window.ethereum.isMetaMask);
+
+      const local = window.localStorage.getItem("metamaskState");
+
+      // user was previously connected, start listening to MM
+      if (local) {
+        listen();
+      }
+
+      // local could be null if not present in LocalStorage
+      const { wallet, balance } = local
+        ? JSON.parse(local)
+        : // backup if local storage is empty
+          { wallet: null, balance: null };
+
+      dispatch({ type: "pageLoaded", isMetamaskInstalled, wallet, balance });
+    }
+  }, []);
+
+  return (
+    <>
+      <Wallet />
+    </>
+  );
+};
+
+export default Home;
+```
+
+In this last page update to `pages/index.tsx` we have relocated the `useEffect` from the `hooks/useMetaMask.tsx` page for separation of concerns since the hook is consuming **dispatch** as a proper next step would to be creating a layout page with NextJS, but since we only have one page we simply added this code here.
```

Comment:
Can you fix this sentence? It doesn't make sense as is.

```suggestion
In this last page update to `pages/index.tsx` we have relocated the `useEffect` from the `hooks/useMetaMask.tsx` page for separation of concerns since the hook is consuming **dispatch** as a proper next step would to be creating a layout page with NextJS, but since we only have one page we simply added this code here.
```